src/shared/domain/definition/case-field-model.spec.ts

```diff
@@ -49,6 +49,69 @@ describe('CaseField', () => {
     component.display_context = 'REAdOnlY';
     expect(component.isReadonly()).toBe(true);
   });
+  it('should be able to test if the field is optional or mandatory', () => {
+    expect(component.isMandatory()).toBe(false);
+    expect(component.isOptional()).toBe(false);
+    component.display_context = null
+    expect(component.isMandatory()).toBe(false);
+    expect(component.isOptional()).toBe(false);
+    component.display_context = 'Mandatory';
+    expect(component.isMandatory()).toBe(true);
+    expect(component.isOptional()).toBe(false);
+    component.display_context = 'Optional';
+    expect(component.isMandatory()).toBe(false);
+    expect(component.isOptional()).toBe(true);
+    component.display_context = 'REAdOnlY';
```

Comment:
_Question:_
Errr... 1337 speak? 🤔

Reply:
This is testing the case changes in the isReadonly() as all letters should be set to the same case in order to equal 'READONLY' so I think it's beneficial. If you disagree, I can edit it to show in a more presentable format?

Reply:
Okay, that's cool... maybe have an `expect(component.isReadonly()).toBe(false);` assertion below it too if it's case-sensitive. That would make it clear why it's there. :)

Reply:
No, the point is that the case is changed so that it's not case sensitive. A .toUpperCase() is used to get the case to be irrelevant

Reply:
Ah, then `expect(component.isReadonly()).toBe(true);` for that one and not for the others! 😂

Reply:
There is a test for it above (I've changed the mention of it here to not be 1337)